pages/content/amp-dev/documentation/guides-and-tutorials/optimize-measure/server-side-rendering.md

```diff
@@ -0,0 +1,74 @@
+
+```html
+<html ⚡ i-amphtml-layout i-amphtml-no-boilerplate transformed="self">
+```
+
+Note: **Note:** AMP caches use a different transformed flag, for example, the Google AMP caches adds `transformed=google;v1`.
+
+&#8291;**2. Remove the AMP boilerplate (when possible):** The [amp-boilerplate code](https://amp.dev/documentation/guides-and-tutorials/learn/spec/amp-boilerplate) has been removed. This means that page content is no longer hidden initially and renders straight away. This greatly increases the perceived performance of an AMP page. It's not always possible to remove the AMP boilerplate. For example, some AMP components, such as `amp-experiment` or `amp-dynamic-css-classes`, modify the page layout based on the user agent, which cannot be server-side rendered. In order to avoid content jumps, these need to be loaded before the page content can be displayed, which requires the AMP boilerplate code to initially hide the content of the page.
+
+```html
+<̶s̶t̶y̶l̶e̶ ̶a̶m̶p̶-̶b̶o̶i̶l̶e̶r̶p̶l̶a̶t̶e̶>̶b̶o̶d̶y̶{̶-̶w̶e̶b̶k̶i̶t̶-̶a̶n̶i̶m̶a̶t̶i̶o̶n̶:̶-̶a̶m̶p̶-̶s̶t̶a̶r̶t̶ ̶8̶s̶ ̶s̶t̶e̶p̶s̶(̶1̶,̶e̶n̶d̶)̶ ̶0̶s̶ ̶1̶ ̶n̶o̶r̶m̶a̶l̶ ̶b̶o̶t̶h̶;̶-̶m̶o̶z̶-̶a̶n̶i̶m̶a̶t̶i̶o̶n̶:̶-̶a̶m̶p̶-̶s̶t̶a̶r̶t̶ ̶8̶s̶ ̶s̶t̶e̶p̶s̶(̶1̶,̶e̶n̶d̶)̶ ̶0̶s̶ ̶1̶ ̶n̶o̶r̶m̶a̶l̶ ̶b̶o̶t̶h̶;̶-̶m̶s̶-̶a̶n̶i̶m̶a̶t̶i̶o̶n̶:̶-̶a̶m̶p̶-̶s̶t̶a̶r̶t̶ ̶8̶s̶ ̶s̶t̶e̶p̶s̶(̶1̶,̶e̶n̶d̶)̶ ̶0̶s̶ ̶1̶ ̶n̶o̶r̶m̶a̶l̶ ̶b̶o̶t̶h̶;̶a̶n̶i̶m̶a̶t̶i̶o̶n̶:̶-̶a̶m̶p̶-̶s̶t̶a̶r̶t̶ ̶8̶s̶ ̶s̶t̶e̶p̶s̶(̶1̶,̶e̶n̶d̶)̶ ̶0̶s̶ ̶1̶ ̶n̶o̶r̶m̶a̶l̶ ̶b̶o̶t̶h̶}̶@̶-̶w̶e̶b̶k̶i̶t̶-̶k̶e̶y̶f̶r̶a̶m̶e̶s̶ ̶-̶a̶m̶p̶-̶s̶t̶a̶r̶t̶{̶f̶r̶o̶m̶{̶v̶i̶s̶i̶b̶i̶l̶i̶t̶y̶:̶h̶i̶d̶d̶e̶n̶}̶t̶o̶{̶v̶i̶s̶i̶b̶i̶l̶i̶t̶y̶:̶v̶i̶s̶i̶b̶l̶e̶}̶}̶@̶-̶m̶o̶z̶-̶k̶e̶y̶f̶r̶a̶m̶e̶s̶ ̶-̶a̶m̶p̶-̶s̶t̶a̶r̶t̶{̶f̶r̶o̶m̶{̶v̶i̶s̶i̶b̶i̶l̶i̶t̶y̶:̶h̶i̶d̶d̶e̶n̶}̶t̶o̶{̶v̶i̶s̶i̶b̶i̶l̶i̶t̶y̶:̶v̶i̶s̶i̶b̶l̶e̶}̶}̶@̶-̶m̶s̶-̶k̶e̶y̶f̶r̶a̶m̶e̶s̶ ̶-̶a̶m̶p̶-̶s̶t̶a̶r̶t̶{̶f̶r̶o̶m̶{̶v̶i̶s̶i̶b̶i̶l̶i̶t̶y̶:̶h̶i̶d̶d̶e̶n̶}̶t̶o̶{̶v̶i̶s̶i̶b̶i̶l̶i̶t̶y̶:̶v̶i̶s̶i̶b̶l̶e̶}̶}̶@̶-̶o̶-̶k̶e̶y̶f̶r̶a̶m̶e̶s̶ ̶-̶a̶m̶p̶-̶s̶t̶a̶r̶t̶{̶f̶r̶o̶m̶{̶v̶i̶s̶i̶b̶i̶l̶i̶t̶y̶:̶h̶i̶d̶d̶e̶n̶}̶t̶o̶{̶v̶i̶s̶i̶b̶i̶l̶i̶t̶y̶:̶v̶i̶s̶i̶b̶l̶e̶}̶}̶@̶k̶e̶y̶f̶r̶a̶m̶e̶s̶ ̶-̶a̶m̶p̶-̶s̶t̶a̶r̶t̶{̶f̶r̶o̶m̶{̶v̶i̶s̶i̶b̶i̶l̶i̶t̶y̶:̶h̶i̶d̶d̶e̶n̶}̶t̶o̶{̶v̶i̶s̶i̶b̶i̶l̶i̶t̶y̶:̶v̶i̶s̶i̶b̶l̶e̶}̶}̶<̶/̶s̶t̶y̶l̶e̶>̶<̶n̶o̶s̶c̶r̶i̶p̶t̶>̶<̶s̶t̶y̶l̶e̶ ̶a̶m̶p̶-̶b̶o̶i̶l̶e̶r̶p̶l̶a̶t̶e̶>̶b̶o̶d̶y̶{̶-̶w̶e̶b̶k̶i̶t̶-̶a̶n̶i̶m̶a̶t̶i̶o̶n̶:̶n̶o̶n̶e̶;̶-̶m̶o̶z̶-̶a̶n̶i̶m̶a̶t̶i̶o̶n̶:̶n̶o̶n̶e̶;̶-̶m̶s̶-̶a̶n̶i̶m̶a̶t̶i̶o̶n̶:̶n̶o̶n̶e̶;̶a̶n̶i̶m̶a̶t̶i̶o̶n̶:̶n̶o̶n̶e̶}̶<̶/̶s̶t̶y̶l̶e̶>̶<̶/̶n̶o̶s̶c̶r̶i̶p̶t̶>̶
+```
+
+
+&#8291;**3. Inline the AMP runtime CSS styles: ** Instead, the AMP runtime CSS ([v0.css](https://cdn.ampproject.org/v0.css)) has been inlined (this gets added by the AMP runtime client-side otherwise). The CSS is required for the AMP layouts to work. To avoid potential version conflicts, the AMP runtime checks the version specified in `i-amphtml-version="011905222334000"` and if it differs from the runtime's version, will update the CSS with the latest version. 
```

Comment:
maybe change to "AMP-internal CSS styles" or something, to get rid of runtime?

Reply:
but runtime might be unavoidable as it's a term in the style tag below, so you could also quickly document what is meant by runtime..

Reply:
Rephrased the section. PTAL